### ISSUE TITLE
ADFS page stuck at waiting

### ISSUE THREAD

**Issue by schrockmatthew** [Yvand/LDAPCP#99]
Not sure what the issue might be........When I sign in using ADFS after the newest ldapcp.wsp install....The sign in page is stuck at waiting on the adfs page before it states problem with "https://url/_trust/".........Not one error in the sharepoint logs and only see fatal communication errors in even viewer on the Windows Process Activation Service

**Comment by Yvand** [comment]
Can you capture a Fiddler trace, confirm it hangs on the POST to /_trust?
If so, then you can get its correlation ID and use it to filter SharePoint logs. It should help you understand the cause of the hang.

**Comment by schrockmatthew** [comment]
Thank you for contacting. Unfortunately, we had to uninstall the ldapcp.wsp and are in the process of building our own people picker via adfs.

In the SharePoint logs it would state authenticated successfully but would eventually time out on the _trust page. It would successfully augment the groups correctly as well. In the windows logs however, it would state a .Net fault in the site that used ldapcp.

Any further help would be appreciated though.

________________________________

**Comment by Yvand** [comment]
Did the problem disappear just after you uninstalled LDAPCP?

**Comment by khalilG** [comment]
We have the same issue here.
The weird thing is that the first time we installed the ldapcp.wsp solution, we tested on an already opened site and it worked for a while. After closing and reopening the browser, problems showed up.
As a workaround, You can use this code to revert back without uninstalling the solution : 
`
![image](https://user-images.githubusercontent.com/8361028/66045704-8b4a5a80-e524-11e9-8091-c822eeb1f83e.png)
`
Also, we'll try an older version, we'll update this post to tell if it worked or not.

**Comment by schrockmatthew** [comment]
Yes but I had to run the sptrusted command again to get it to work as it was before.

________________________________

**Comment by Yvand** [comment]
Can you also make sure that you visit a LDAPCP administration page (hosted in central administration > security) before you try to load any site.
The reason is that visiting a LDAPCP page will create a configuration object, which may fix this issue

**Comment by khalilG** [comment]
@Yvand Yes, I visited the administration page on the central administration and, by the way, launched the configuration script available here : [](https://ldapcp.com/Configure-LDAPCP.html)

**Comment by schrockmatthew** [comment]
Ldapcp states in the SharePoint log that configuration is successful. The logs even state successful token and augment of user groups. I believe it’s .Net issue.

________________________________

**Comment by khalilG** [comment]
When I try to debug on our test environment, we got this issue :
![image](https://user-images.githubusercontent.com/8361028/66114914-0f0d5100-e5d0-11e9-89f0-48b9030fd451.png)
System.StackOverflowException
But no more explanation.

**Comment by khalilG** [comment]
I finally found a way to deploy the solution and attach breakpoints.
What is remarkable is that it keeps hitting the "Initialize" function again and again until StackOverflow exception is raised.
![image](https://user-images.githubusercontent.com/8361028/66128958-37587800-e5ef-11e9-94cc-b78719bd70cd.png)
@Yvand : any idea why it's keeping initializing ?
Thanks.

**Comment by Yvand** [comment]
In Visual Studio > Debug > Options: can you unckeck "Enable just my code" and break again?
It should show the SharePoint frames

**Comment by khalilG** [comment]
> In Visual Studio > Debug > Options: can you unckeck "Enable just my code" and break again?
> It should show the SharePoint frames

Will try that now !

**Comment by khalilG** [comment]
Here are some screenshots : 

![image](https://user-images.githubusercontent.com/8361028/66129598-8357ec80-e5f0-11e9-8b6d-3a1d4fea0cc1.png)

![image](https://user-images.githubusercontent.com/8361028/66129606-86eb7380-e5f0-11e9-804d-735f21a392c3.png)

**Comment by khalilG** [comment]
As described in the Call Stack, the method "GetUerKeyForEntity" is keeping called again and again. After debugging it, I found that some breakpoints are never reached, as described below 👍 
![image](https://user-images.githubusercontent.com/8361028/66132540-81dcf300-e5f5-11e9-9ca2-dfbbfc259a9c.png)

**Comment by Yvand** [comment]
I see, the StackOverflow is caused by the elevation of privileges done in method GetUserKeyForEntity.
As you can see in the comment, this elevation of privileges was added in v14 to fix issue https://github.com/Yvand/LDAPCP/issues/87, unfortunately, in your environment this is causing the StackOverflow, which is bad news.
Can you let me know:
- What is your SharePoint build?
- The configuration of your web app (zones / authentication providers enabled per zone)

**Comment by khalilG** [comment]
- Farm Build Version :  16.0.4717.1000
- Web App configuration : 
![image](https://user-images.githubusercontent.com/8361028/66137070-d2a41a00-e5fc-11e9-87d3-8605b252fb08.png)
![image](https://user-images.githubusercontent.com/8361028/66137122-e780ad80-e5fc-11e9-9e6f-b6da3ab68c77.png)
![image](https://user-images.githubusercontent.com/8361028/66137148-f10a1580-e5fc-11e9-8fe8-331df6840f00.png)

**Comment by Yvand** [comment]
Can you revert your customization of the default sign in page to default sign in page and confirm that you still reproduce the issue?

**Comment by khalilG** [comment]
I revert back to the default sign in page. When I open my site, I have 2 choices : 
![image](https://user-images.githubusercontent.com/8361028/66139934-8c04ee80-e601-11e9-9712-ea6235a342db.png)

If I choose 'Windows Authentication', everything works fine, site is accessible and I can get AD users from the people picker : 
![image](https://user-images.githubusercontent.com/8361028/66140005-a63ecc80-e601-11e9-88ef-a27440248d09.png)

But if I choose our trusted identity provider, the error occurs : 
![image](https://user-images.githubusercontent.com/8361028/66140044-b656ac00-e601-11e9-8d23-e1bf05e3bba5.png)

**Comment by Yvand** [comment]
So it is the same issue, assuming this unhandled exception is the StackOverflow caused by the elevation of privileges in GetUserKeyForEntity.

I added this elevation of privileges as part of a larger change to fix issue https://github.com/Yvand/LDAPCP/issues/87 and I never reproduced this StackOverflow during my tests, but you are 2 to reportedly reproduce it.
Considering this, I will revert the change and publish soon an update with this sole change.

Issue https://github.com/Yvand/LDAPCP/issues/87 is still mitigated by other changes also introduced in current version in v14, so it's not that bad.

**Comment by kalervojankko** [comment]
> Not sure what the issue might be........When I sign in using ADFS after the newest ldapcp.wsp install....The sign in page is stuck at waiting on the adfs page before it states problem with "[https://url/_trust/".........Not](https://url/_trust/%22.........Not) one error in the sharepoint logs and only see fatal communication errors in even viewer on the Windows Process Activation Service

Had exactly the same issue right after I ran the Update-SPSolution command as I updated older LDAPCP version to this version.

After two hours of troubleshooting I found this post and then decided to remove LDAPCP with these instructions: https://ldapcp.com/Remove-LDAPCP.html and then installed version 13.0.20190621.905 and everything worked again. I already installed earlier this 13. version to our TEST farm so I knew this shouldn't have any issues.

**Comment by khalilG** [comment]
> > Not sure what the issue might be........When I sign in using ADFS after the newest ldapcp.wsp install....The sign in page is stuck at waiting on the adfs page before it states problem with "[https://url/_trust/".........Not](https://url/_trust/%22.........Not) one error in the sharepoint logs and only see fatal communication errors in even viewer on the Windows Process Activation Service
> 
> Had exactly the same issue right after I ran the Update-SPSolution command as I updated older LDAPCP version to this version.
> 
> After two hours of troubleshooting I found this post and then decided to remove LDAPCP with these instructions: https://ldapcp.com/Remove-LDAPCP.html and then installed version 13.0.20190621.905 and everything worked again. I already installed earlier this 13. version to our TEST farm so I knew this shouldn't have any issues.

@kalervojankko : will try this Monday and let you know if it succeeded.

**Comment by khalilG** [comment]
@kalervojankko I just installed the 13.0.20190621.905 version and it seems to be perfectly working.

**Comment by Yvand** [comment]
@khalilG @kalervojankko this is clearly a regression in v14, I will publish soon v14.1 to fix it, code is ready and I'm running integration tests on it.
Thank you for reporting and updating me with your tests and findings

**Comment by Yvand** [comment]
@khalilG, @kalervojankko, @schrockmatthew, are you willing to test [v14.1 beta](https://github.com/Yvand/LDAPCP/releases/tag/14.1.20191007.981) and confirm that it resolves the issue?
As I'm not able to repro it, it would be great to have your validation before I publish it.

**Comment by kalervojankko** [comment]
@Yvand I can test it tomorrow in our TEST farm and will let you know how it goes.

**Comment by khalilG** [comment]
> @khalilG, @kalervojankko, @schrockmatthew, are you willing to test [v14.1 beta](https://github.com/Yvand/LDAPCP/releases/tag/14.1.20191007.981) and confirm that it resolves the issue?
> As I'm not able to repro it, it would be great to have your validation before I publish it.

@Yvand Will do that today and let you know if it succeeded.

**Comment by kalervojankko** [comment]
> @khalilG, @kalervojankko, @schrockmatthew, are you willing to test [v14.1 beta](https://github.com/Yvand/LDAPCP/releases/tag/14.1.20191007.981) and confirm that it resolves the issue?
> As I'm not able to repro it, it would be great to have your validation before I publish it.

Ok, I have now installed this 14.1.20191007.981 beta version and seems to work. I didn't make any comprehensive testing, but at least it didn't break. :)

**Comment by Yvand** [comment]
@kalervojankko that's very good news, many thanks for taking the time to test it.
@khalilG I'll wait for your feedback. If it is positive, I'll make the pre-release a normal release, so it will be the version downloaded by default

**Comment by khalilG** [comment]
> @kalervojankko that's very good news, many thanks for taking the time to test it.
> @khalilG I'll wait for your feedback. If it is positive, I'll make the pre-release a normal release, so it will be the version downloaded by default

@Yvand I've got some issues with my SharePoint test environment, nothing related to LDAPCP, so seems like I'll not be able to give you a feedback for today, hope things will get better and try to test tomorrow. Thanks.

**Comment by Yvand** [comment]
@khalilG thanks for the update, please do let me know if you are not able to complete the test by tomorrow

**Comment by khalilG** [comment]
@Yvand Sorry but I don't think I'll be able to give you feedback today, I'm stuck with another issue. I think if it works for the others, it will be the same for me.
Thank you again for your efforts.

**Comment by Yvand** [comment]
@AmitW02 this looks like another thread took the lock and does not release it.
What version of LDAPCP.dll exactly are you debugging ?

**Comment by Yvand** [comment]
@AmitW02 is this issue still present?

**Comment by schrockmatthew** [comment]
Not for I.
________________________________

**Comment by AmitW02** [comment]
I am good now. Thanks.
